src/mango.ts

```diff
@@ -301,38 +316,38 @@ export async function gammaScalpMango(
     ? Math.max(0.01, dipTotalGamma * stdDevSpread * fairValue)
     : dipTotalGamma * stdDevSpread * fairValue;
 
-  if (netGamma < scalper.minSize) {
-    console.log(scalper.symbol, 'Gamma Hedge Too Small', netGamma, 'vs', scalper.minSize);
+  const gammaOrderQty = roundQtyToMinOrderStep(netGamma, scalper.minSize);
+  if (gammaOrderQty < scalper.minSize) {
+    console.log(scalper.symbol, 'Gamma Hedge Too Small', gammaOrderQty, 'vs', scalper.minSize);
     return;
   }
-  console.log(scalper.symbol, 'Position Gamma Γ:', netGamma, 'Fair Value', fairValue);
+  console.log(scalper.symbol, 'Position Gamma Γ:', gammaOrderQty, 'Fair Value', fairValue, 'Scalp #', gammaScalpCount);
 
   const orderIdGamma = new Date().getTime() * 2;
-  const gammaBid = fairValue * (1 - stdDevSpread);
+  const gammaBid = roundPriceToTickSize(fairValue * (1 - stdDevSpread), scalper.tickSize);
   const gammaBidID = orderIdGamma + 1;
-  const gammaAsk = fairValue * (1 + stdDevSpread);
+  const gammaAsk = roundPriceToTickSize(fairValue * (1 + stdDevSpread), scalper.tickSize);
   const gammaAskID = orderIdGamma + 2;
 
   let gammaFillQty = 0;
-  // TODO: Check fills feed works here
   fillFeed.removeAllListeners('message');
-  const gammaFillListener = (event) => {
-    const parsedEvent = JSON.parse(event.data);
+  const gammaFillListener = (event: WebSocket.MessageEvent) => {
+    const parsedEvent = JSON.parse(event.data as string);
     const {
-      owner, clientOrderId, quantity, price,
+      maker, makerClientOrderId, quantity, price,
     } = parsedEvent.event;
-    if (owner !== mangoAccount.publicKey.toBase58()) {
+    if (maker !== mangoAccount.publicKey.toBase58()) {
       return;
     }
-    if (clientOrderId === gammaBidID || clientOrderId === gammaAskID) {
+    if (makerClientOrderId === gammaBidID || makerClientOrderId === gammaAskID) {
       gammaFillQty += Math.abs(quantity);
       // Once the gamma fills have crossed the threshold, reset the orders.
-      if (gammaFillQty > netGamma * GAMMA_COMPLETE_THRESHOLD_PCT) {
+      if (gammaFillQty > gammaOrderQty * GAMMA_COMPLETE_THRESHOLD_PCT) {
```

Comment:
The fill objects were changed to be just 1 per trade event rather than 1 taker and 1 maker. The changes here to netGamma are just to round to the min step size so orders don't get rejected. No logic changes

Reply:
This all should be documented somewhere. Link to the new spec ideally.
You are using different parameters, so it is a logic change, even if it seems safe.